apps/dashboard/src/components/number-input-with-select.tsx

```diff
@@ -0,0 +1,104 @@
+import { cn } from '@/utils/ui';
+import { FormControl, FormField, FormItem, FormMessagePure } from '@/components/primitives/form/form';
+import { Input } from '@/components/primitives/input';
+import { InputFieldPure } from '@/components/primitives/input';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { useFormContext } from 'react-hook-form';
+import { useMemo } from 'react';
+
+type InputWithSelectProps = {
+  fields: {
+    inputKey: string;
+    selectKey: string;
+  };
+  options: string[];
+  defaultOption?: string;
+  className?: string;
+  placeholder?: string;
+  isReadOnly?: boolean;
+};
+
+export const NumberInputWithSelect = ({
+  fields,
+  options,
+  defaultOption,
+  className,
+  placeholder,
+  isReadOnly,
+}: InputWithSelectProps) => {
+  const { getFieldState, setValue, getValues, control } = useFormContext();
```

Comment:
The wrapper component already exists. See [here](https://github.com/novuhq/novu/pull/7131/files#diff-2232d48a43ebf90e611f34a4f12b354dc571e6640164e03a8d62f78ce4950b76R1).

Reply:
I'm not sure we can do something about it because this component uses the two form fields, check the `FormField` usage.

Reply:
Lets see how it evolves with the digest and we will incrementally adjust this component, by either making it primitive or whatever we need - I dont want to make it too generic too early since I dont know what will digest need.

Reply:
Folks, I strongly disagree with this approach. We start by pairing all our components with a context, making them stateful, whereas it is obvious they should be stateless, with presentational components wired to their state on each page.

I think of them as our internal UI component library, which is a composition of primitives and composite stateless components. This number input with Select is a composite.

Reply:
Tend to agree with Adam here, we tend to overengineer a lot lately without the need, i would say whoever will implement Digest should be aware of Delay components and refactor accordingly